src/deepsparse/pipeline.py

```diff
@@ -167,11 +175,29 @@ def __init__(
         self.onnx_file_path = self.setup_onnx_file_path()
         self.engine = self._initialize_engine()
 
-    def __call__(self, threadpool=None, *args, **kwargs) -> Union[BaseModel, Future]:
-        threadpool = threadpool or self._threadpool
-        if threadpool:
-            return threadpool.submit(self._run, *args, **kwargs)
-        return self._run(*args, **kwargs)
+    def __call__(self, *args, **kwargs) -> Union[BaseModel, Future]:
+        executor = kwargs.get("executor", _NOT_PRESENT_FLAG)
+
+        if executor is None:
+            # do not use threading
+            return self._run(*args, **kwargs)
+
+        if executor is _NOT_PRESENT_FLAG:
+            # use executor created during initialization
+            executor = self.executor
+        else:
+            # use passed in executor
+            executor = kwargs.pop("executor")
+
+        # if self.use_dynamic_batch():
+        #     # Blocking call in Dynamic Batch Mode
+        #     return executor.submit(self._run, *args, **kwargs).result()
```

Comment:
nit: prefer to avoid merging  blocks of commented code

Reply:
Reviewing code b/w commits I see 
<img src="https://media0.giphy.com/media/xUOxeZUc8UFwMgH2MM/giphy-downsized-medium.gif"/>